app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/event_catcher/stream.rb

```diff
@@ -0,0 +1,34 @@
+class ManageIQ::Providers::IbmCloud::VPC::CloudManager::EventCatcher::Stream
+  attr_reader :ems, :stop_polling, :poll_sleep
+
+  def initialize(ems, options = {})
+    @ems = ems
+    @stop_polling = false
+    @from = DateTime.current.to_time.to_i.to_s
+    @poll_sleep = options[:poll_sleep] || 20.seconds
+  end
+
+  def start
+    @stop_polling = false
+    @from = DateTime.current.to_time.to_i.to_s
+  end
+
+  def stop
+    @stop_polling = true
+  end
+
+  def poll
+    loop do
+      @to = DateTime.current.to_time.to_i.to_s
+
+      ems.with_provider_connection(:service => 'events') do |api|
+        # Replace with your provider's events API
+        events_client = api.sdk_client
+        events = events_client.export(:from => @from, :to => @to)
```

Comment:
Think you need to `yield events` here, actually the caller is expecting one event at a time so `events.each { |e| yield e }` or take a `&block` like rubocop was recommending and to `events.each(&block)`

Reply:
`events` returned from the API is a `IBMCloudSdkCore::DetailedResponse` object where the response is a single string containing all the events that occurred during the `@from -> @to` timeframe. Therefore I am not able to use `.each` here. Let me know if my understanding is incorrect or if there is another way.

i.e.
```
#<IBMCloudSdkCore::DetailedResponse:0x00007fdbfbca21d0 @status=200, @headers={"Server"=>"restify", "Content-Security-Policy"=>"frame-ancestors 'none'", "Strict-Transport-Security"=>"max-age=31536000; includeSubDomains", "Date"=>"Thu, 29 Jul 2021 15:59:47 GMT", "Connection"=>"close", "Transfer-Encoding"=>"chunked"}, @result="{\"_account\":\"c6e1d98061\",\"_cluster\":\"ld86\",\"_host\":\"containers-kubernetes\",\"_ingester\":\"supertenant\",\"_logtype\":\"json\",\"_file\":\"crn:v1:bluemix:public:containers-kubernetes:us-east:a/c56c9a268d23e1b339ac14774358133c:c31pbtiw0gfedg1bna4g::\",\"_line\":\"{\\\"action\\\":\\\"containers-kubernetes.cluster-ingress-status.get\\\",\\\"correlationId\\\":\\\"ff9f6161-9c2d-4cba-b6df-81581d05d580\\\",\\\"dataEvent\\\":true,\\\"eventTime\\\":\\\"2021-07-29T15:41:52.29+0000\\\",\\\"initiator\\\":{\\\"id\\\":\\\"crn:v1:bluemix:public:containers-kubernetes\\\",\\\"name\\\":\\\"armada-alb-api\\\",\\\"typeURI\\\":\\\"service/security/account/service\\\",\\\"host\\\":{\\\"address\\\":\\\"\\\",\\\"addressType\\\":\\\"\\\"},\\\"credential\\\":{\\\"type\\\":\\\"\\\"}},\\\"logSourceCRN\\\":\\\"crn:v1:bluemix:public:containers-kubernetes:us-east:a/c56c9a268d23e1b339ac14774358133c:c31pbtiw0gfedg1bna4g::\\\",\\\"message\\\":\\\"Kubernetes Service: get cluster-ingress-status c31pbtiw0gfedg1bna4g\\\",\\\"observer\\\":{\\\"name\\\":\\\"ActivityTracker\\\"},\\\"outcome\\\":\\\"success\\\",\\\"reason\\\":{\\\"reasonCode\\\":200,\\\"reasonType\\\":\\\"OK\\\",\\\"reasonForFailure\\\":\\\"\\\"},\\\"resourceGroupID\\\":\\\"crn:v1:bluemix:public:resource-controller::a/c56c9a268d23e1b339ac14774358133c::resource-group:345c433098294722ba52d9039133e8cf\\\",\\\"saveServiceCopy\\\":true,\\\"severity\\\":\\\"normal\\\",\\\"target\\\":{\\\"id\\\":\\\"crn:v1:bluemix:public:containers-kubernetes:us-east:a/c56c9a268d23e1b339ac14774358133c:c31pbtiw0gfedg1bna4g::\\\",\\\"name\\\":\\\"c31pbtiw0gfedg1bna4g\\\",\\\"typeURI\\\":\\\"containers-kubernetes/cluster\\\",\\\"host\\\":{},\\\"resourceGroupId\\\":\\\"crn:v1:bluemix:public:resource-controller::a/c56c9a268d23e1b339ac14774358133c::resource-group:345c433098294722ba52d9039133e8cf\\\"},\\\"id\\\":\\\"ff9f6161-9c2d-4cba-b6df-81581d05d580\\\"}\",\"_rawline\":null,\"level\":\"normal\",\"_ts\":1627574361281,\"_platform\":\"containers-kubernetes\",\"_app\":\"crn:v1:bluemix:public:containers-kubernetes:us-east:a/c56c9a268d23e1b339ac14774358133c:c31pbtiw0gfedg1bna4g::\",\"_ip\":\"166.9.14.3\",\"__key\":\"logline:c6e1d98061:crn:v1:bluemix:public:containers-kubernetes:us-east:a/c56c9a268d23e1b339ac14774358133c:c31pbtiw0gfedg1bna4g:::containers-kubernetes:ld86\",\"_bid\":\"852a8e56-6e9a-4c7b-8671-c6b3ef2cabe5:1302:ld86\",\"action\":\"containers-kubernetes.cluster-ingress-status.get\",\"correlationId\":\"ff9f6161-9c2d-4cba-b6df-81581d05d580\",\"dataEvent\":true,\"eventTime\":\"2021-07-29T15:41:52.29+0000\",\"logSourceCRN\":\"crn:v1:bluemix:public:containers-kubernetes:us-east:a/c56c9a268d23e1b339ac14774358133c:c31pbtiw0gfedg1bna4g::\",\"message\":\"Kubernetes Service: get cluster-ingress-status c31pbtiw0gfedg1bna4g\",\"outcome\":\"success\",\"resourceGroupID\":\"crn:v1:bluemix:public:resource-controller::a/c56c9a268d23e1b339ac14774358133c::resource-group:345c433098294722ba52d9039133e8cf\",\"saveServiceCopy\":true,\"severity\":\"normal\",\"id\":\"ff9f6161-9c2d-4cba-b6df-81581d05d580\",\"_id\":\"1373946466414157825\",\"initiator\":{\"id\":\"crn:v1:bluemix:public:containers-kubernetes\",\"name\":\"armada-alb-api\",\"typeURI\":\"service/security/account/service\",\"host\":{\"address\":\"\",\"addressType\":\"\"},\"credential\":{\"type\":\"\"}},\"observer\":{\"name\":\"ActivityTracker\"},\"reason\":{\"reasonCode\":200,\"reasonType\":\"OK\",\"reasonForFailure\":\"\"},\"target\":{\"id\":\"crn:v1:bluemix:public:containers-kubernetes:us-east:a/c56c9a268d23e1b339ac14774358133c:c31pbtiw0gfedg1bna4g::\",\"name\":\"c31pbtiw0gfedg1bna4g\",\"typeURI\":\"containers-kubernetes/cluster\",\"resourceGroupId\":\"crn:v1:bluemix:public:resource-controller::a/c56c9a268d23e1b339ac14774358133c::resource-group:345c433098294722ba52d9039133e8cf\",\"host\":{}}}\n">
```